### PR TITLE
#159053959 Enable social login via google or facebook.

### PR DESCRIPTION
**Pull request by chepmanu**
#### What does this PR do?
Setup social authentication using Google and Facebook.

#### Description of tasks to be completed
1.Configure the social auth settings in the settings file.  
2.Go to the Oauth providers and register the application.
3.You can use mock ups from the providers to get access token for testing out the functionalities.
#### How should this be manually tested?
```sh
$ git clone https://github.com/andela/ah-titans.git
```
```sh
$ git checkout -b ft-enable-social-login-via-google-159053959
```
```sh
$ cd ah-titans/authors
```

> Add **.env** file and copy **example.env** content.
> Add your application's I.D and secret to the  **.env** file

```sh
$ source .env
```
```sh
$ python3.6 -m venv venv
```
```sh
$ source venv/bin/activate
```
```sh
$ pip install -r requirements.txt
```
```sh
$ python manage.py makemigrations
```
```sh
$ python manage.py runserver
```

#### Any background context you want to provide?
The feature only handles a small part of an entire process that starts with the front-end part.

#### What are the relevant Pivotal Tracker Stories?
[#159053959](https://www.pivotaltracker.com/story/show/159053959)

### CODE REVIEW

**Review by petermwash — APPROVED** [review]

**Comment by janetnim** [comment]
Add end of line in the files and remove details entailing twitter as you are not using it anymore

**Comment by evansmusomi** [comment]
@chepmanu, the format of your PR title isn't inline with the convention used for other PRs. Please edit it. 

In the section on what this PR does, you didn't include Twitter although it's part of the PT story title - was that intentional?

**Comment by chepmanu** [comment]
**@evansmusomi**, The choice of not including Twitter is intentional since I have not had the app registered yet(Twitter is still reviewing my application), I was therefore asked to push what I had so that it can be reviewed as I wait to implement on the twitter issue. Note taken concerning the PR naming convention.

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -116,3 +126,60 @@ def update(self, request, *args, **kwargs):
...
+                return Response(
+                    {'errors': {"user": 'This user account is inactive'}},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+        else:
+            return Response(
+                {'errors': {"user": "Authentication Failed"}},
+                status=status.HTTP_400_BAD_REQUEST,
+
+            )
```
Comment (petermwash) [inline authors/apps/authentication/views.py]: Add an end of line here.

File: authors/settings.py [hunk 2]
```diff
@@ -152,3 +158,52 @@
...
+    'social_core.pipeline.social_auth.associate_by_email',  
+    'social_core.pipeline.user.create_user',
+    'social_core.pipeline.social_auth.associate_user',
+    'social_core.pipeline.social_auth.load_extra_data',
+    'social_core.pipeline.user.user_details',
+)
+
+SOCIAL_AUTH_ADMIN_USER_SEARCH_FIELDS = ['username', 'first_name', 'email']
+SOCIAL_AUTH_USER_FIELDS = ['email', 'username']
+SOCIAL_AUTH_URL_NAMESPACE = 'social'
```
Comment (petermwash) [inline authors/settings.py]: Add an end of line.

**Pull request merged**